code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -126,10 +126,11 @@
 		return src.close()
 	return src.open()
 
-// this should probably use dump_contents()
 /obj/structure/closet/ex_act(severity, target)
 	contents_explosion(severity, target)

```

Comment:
I'm pretty sure the reason the closet's content is getting deleted is because contents_explosion() is called. If you keep it in, it doesn't fix 7376.